ambry-api/src/main/java/com.github.ambry/rest/RequestPath.java

```diff
@@ -50,6 +49,15 @@
    */
   public static RequestPath parse(RestRequest restRequest, List<String> prefixesToRemove, String clusterName)
       throws RestServiceException {
+    String blobIdHeader = RestUtils.getHeader(restRequest.getArgs(), Headers.BLOB_ID, false);
+    if (blobIdHeader != null) {
+      // If blob id is specified in the rest request header and it has cluster name prefix, we remove its prefix and
+      // use this id (with no prefix now) to update "x-ambry-blob-id" header. The purpose is to ensure IdConverter and
+      // IdSigningService can receive a valid blob id and correctly identify it.
+      int nextOffset = matchPathSegments(blobIdHeader, 0, clusterName, true);
+      String blobIdStr = nextOffset >= 0 ? blobIdHeader.substring(nextOffset) : blobIdHeader;
```

Comment:
Could you add a test for this to RequestPathTest?

Reply:
will do

Reply:
Let's do it this way so that it is consistent with signedChunkId: 
```
    String blobIdHeader = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.BLOB_ID, false);
    if (blobIdHeader != null) {
      String blobIdStr =
          parse(blobIdHeader, Collections.emptyMap(), prefixesToRemove, clusterName).getOperationOrBlobId(false);
      restRequest.setArg(RestUtils.Headers.BLOB_ID, blobIdStr);
    }
```
This will also ensure sub resources are stripped off, which our code can't handle